Gems/AWSClientAuth/Code/Tests/AWSClientAuthGemMock.h

```diff
@@ -423,13 +423,14 @@ namespace AWSClientAuthUnitTest
         MOCK_METHOD1(OnSignOut, void(const AWSClientAuth::ProviderNameEnum& providerName));
 
     private:
-        void AssertAuthenticationTokensPopulated(const AWSClientAuth::AuthenticationTokens& authenticationToken)
+        void AssertAuthenticationTokensPopulated([[maybe_unused]] const AWSClientAuth::AuthenticationTokens& authenticationToken)
         {
             AZ_Assert(
                 authenticationToken.GetAccessToken() == TEST_ACCESS_TOKEN,
                 "Access token expected to match");
             AZ_Assert(
-                authenticationToken.GetOpenIdToken() == TEST_ID_TOKEN,
+                authenticationToken.GetProviderName() == AWSClientAuth::ProviderNameEnum::LoginWithAmazon ?
+                authenticationToken.GetOpenIdToken() == TEST_ACCESS_TOKEN : authenticationToken.GetOpenIdToken() == TEST_ID_TOKEN,
```

Comment:
nit: could we indent this line to help show it's part of the preceding ternary operation?

Reply:
Done.